src/Simulation/QSharpFoundation/Convert/Convert.qs

```diff
@@ -94,7 +107,23 @@ namespace Microsoft.Quantum.Convert {
     /// let bi2 = BoolArrayAsBigInt([false, false, false, false, false, false, false, true]); // Not padded -> -128
     /// ```
     function BoolArrayAsBigInt(a : Bool[]) : BigInt {
-        body intrinsic;
+        mutable val = 0L;
```

Comment:
How many bits does `val` have?
What does the user expect if the passed array is longer than that number of bits? And what will really happen?

Reply:
`BigInt` is backed by expanding storage, so should expand to whatever size is needed. This is different than `BoolArrayAsInt` which will fail if the passed array is bigger than 64 entries (see https://github.com/microsoft/QuantumLibraries/blob/main/Standard/src/Convert/Convert.qs#L103-L104).

Reply:
I assume you plan to refactor `BoolArrayAsBigInt()`.